modules/nf-core/blast/blastp/tests/main.nf.test

```diff
@@ -9,19 +9,119 @@ nextflow_process {
+                input[1] = BLAST_MAKEBLASTDB.out.db
+                input[2] = ''
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert path(process.out.tsv.get(0).get(1)).md5 == "dd58eb09f38e6f951aa00c133a18ef5b" },
+                { assert process.out.versions }
+            )
+        }
+
+    }
+
+    test("Should search for zipped proteins against a blast db and return the default (tsv) file with hits") {
 
-        setup {
-            run("BLAST_MAKEBLASTDB") {
-                script "../../makeblastdb/main.nf"
-                process {
-                    """
-                    input[0] = [ file(params.test_data['sarscov2']['genome']['proteome_fasta'], checkIfExists: true) ]
-                    """
-                }
+        when {
+            params {
+                outdir = "$outputDir"
+            }
+            process {
+                """
+                input[0] = [ [id:'test'], file(params.test_data['sarscov2']['genome']['proteome_fasta_gz'], checkIfExists: true) ]
+                input[1] = BLAST_MAKEBLASTDB.out.db
+                input[2] = ''
+                """
             }
         }
 
+        then {
+            assertAll(
+                { assert process.success },
+                { assert path(process.out.tsv.get(0).get(1)).md5 == "dd58eb09f38e6f951aa00c133a18ef5b" },
+                { assert process.out.versions }
+            )
+        }
+
+    }
+
+    test("Should search for proteins against a blast db and return an xml file with hits") {
+
+        when {
+            params {
+                outdir = "$outputDir"
+            }
+            process {
+                """
+                input[0] = [ [id:'test'], file(params.test_data['sarscov2']['genome']['proteome_fasta'], checkIfExists: true) ]
+                input[1] = BLAST_MAKEBLASTDB.out.db
+                input[2] = 'xml'
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert path(process.out.xml.get(0).get(1)).md5 == "e0119340bba8b434c250ffb599eb33a2" },
+                { assert process.out.versions }
+            )
+        }
+
+    }
+
+    test("Should search for proteins against a blast db and return a tsv file with hits") {
+
+        when {
+            params {
+                outdir = "$outputDir"
+            }
+            process {
+                """
+                input[0] = [ [id:'test'], file(params.test_data['sarscov2']['genome']['proteome_fasta'], checkIfExists: true) ]
+                input[1] = BLAST_MAKEBLASTDB.out.db
+                input[2] = 'tsv'
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert path(process.out.tsv.get(0).get(1)).md5 == "dd58eb09f38e6f951aa00c133a18ef5b" },
```

Comment:
Can we use `snapshot` instead? Then we don't have to update the md5s by hand. You can pass just this file.

Reply:
Done